sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1156,6 +1156,99 @@ static void _configure_netdev(netdev_t *dev)
 #endif
 }
 
+#ifdef DEVELHELP
+/* checks if a device supports all required options and functions */
+static void _test_options(gnrc_netif_t *netif)
+{
+#if (GNRC_NETIF_L2ADDR_MAXLEN > 0) || GNRC_IPV6_NIB_CONF_6LN
```

Comment:
```suggestion
#if ((GNRC_NETIF_L2ADDR_MAXLEN > 0) && defined(MODULE_GNRC_SIXLOWPAN_ND)) || GNRC_IPV6_NIB_CONF_6LN
```

or maybe just use

``` C
uint8_t dummy_addr[GNRC_NETIF_L2ADDR_MAXLEN];
(void) dummy_addr;
```

Without and `#if`... `#endif` magic and let the compiler collect it when not used.

Reply:
(To elaborate on that: Murdock complains that `dummy_addr` is in many cases declared, but not used. I think the new `#if` ... `#else` magic I suggest should only declare it when it is actually used - but I did not check it.)

Reply:
The ifdef is wrong... It should be

```suggestion
#if ((GNRC_NETIF_L2ADDR_MAXLEN > 0) && GNRC_IPV6_NIB_CONF_6LN) || \
    MODULE_GNRC_SIXLOWPAN_ND
```

But since the `(void)dummy_addr` doesn't seem to have any implications code-sizewise, I go for that.